### ISSUE TITLE
Question regarding future commits to project.

### ISSUE THREAD

**Issue by mjr4077au** [PSAppDeployToolkit/PSAppDeployToolkit#869]
Hi there, this one's a bit more left field than a normal "how do I do x" question, but more to do with the project on a technical level.

I've known about the toolkit for some time and have a fork of it with some specific changes for our business's requirements. Historically, there's been some incredibly large commits, most recently with ee5fd4771259546aa6982d49988ff4141577fed1 which are almost re-writes of the project itself.

Are commits of this size something I should expect to see in future works against the project, or is there an intention of committing changes on a change-by-change basis to ease merge conflicts and other problems that would arise when merging in a remote-tracking branch?

**Comment by mjr4077au** [comment]
Just to be clear, I don't mean to sound disrespectful or unappreciative of the incredibly hard work that's gone into PSADT; this is just a real-world concern I have that by the time 3.10, or 4.0 comes around, that it could be incredibly hard to take in on my end.

**Comment by sintaxasn** [comment]
Hey Mitchell,

The commit you referenced was a one off - I don't think any of us want to go through that again. So absolutely - change by change tracking is the intended path forward. However, there are some planned changes to the project file structure for v4 which will impact any fork and will require this level of review and merge. 

So, I was going to ask for the reason you had a fork in the first place, and potentially, to suggest if your changes could be folded back in. But based on the number of PRs you've submitted in the last few hours, it seems you've concluded on this approach too. Will start reviewing tomorrow.

**Comment by mjr4077au** [comment]
Thank you for coming back to me. That's good to hear about future commits moving forward! Completely understandable that a major release version like a 4.0 would justifiably require extra attention.

I don't want to maintain a fork at all if I can avoid it, and depending on whether my stuff I've submitted is acceptable, I can probably get my change count on my side close to zero. One of the biggest things is that NonInteractive mode sets `$deployModeSilent` to $true, when in my mind, NonInteractive should be automated interactivity, therefore show things to the user albeit without any prompts. So as such, I set `$deployModeSilent` to `$true` when running non-interactively. To that end, what I've been thinking I can do is probably reset `$deployModeSilent` how I wish in `Deploy-Application.ps1` after dot-sourcing `AppDeployToolkitMain.ps1` and therefore achieve the desired outcome.

Aside from that, #884 and #886 are big ones, as well as being able to control `$dirFiles`. From the PRs, https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/pull/881/commits/e4e569213224135d094adb60dc433c16e006f289 is my only crucial change, the rest are due to my preference of setting `Set-StrictMode -Version Latest` on everything I do, which resulted in a few terminating issues due to undefined variables, etc.
